imagebuildah/stage_executor.go

```diff
@@ -1302,6 +1302,24 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 				if err != nil {
 					return "", nil, fmt.Errorf("checking if cached image exists from a previous build: %w", err)
 				}
+				// All the best effort to find cache on localstorage have failed try pulling
+				// cache from remote repo if `--cache-from` was configured and cacheKey was
```

Comment:
should it be "cacheKey" or "cacheID"?

Reply:
`cacheKey` is the hash or key generated for a particular intermediate image as also used as tag when pushed to remote repo, the keyword was introduced here: https://github.com/containers/buildah/pull/4118

I think difference between `key` and  `id` is , the `ID` is always unique but `key` can be same if the same intermediate image is built again.

Reply:
The ID is an image ID, which is usually the digest of that image's config blob.  That config blob includes data which we don't consider relevant when evaluating whether or not a given image should be considered a cache hit.